Styling

kiddi opened the pull request:
- Flexbox for responsiveness
- moved components out of App.js to simplify
- introducing BEM CSS conventions

kiddi committed:
added flexbox styling and more, wip

kiddi committed:
stuff

johannth commented on src/App.css:
```diff
@@ -1,5 +1,83 @@
+  line-height: 19px;
+  font-family: AvenirLTStd-Black, arial, sans-serif;
+  -webkit-box-sizing: border-box;
+  -moz-box-sizing: border-box;
```
I think create-react-app does some processing so we don't need to use browser specific properties.

kiddi replied:
Yeah, that's very likely.

johannth commented on src/App.js:
```diff
@@ -178,19 +169,21 @@ const DonateFormPage1 = ({ baseAmount, handleInputChange }) => {
+              </div>
+            <div className="DonateFormPage1__CTA">
+              <Link to="/donate/step/2">Next</Link>
+            </div>
```
If you're using atom I would install https://github.com/prettier/prettier. It reformats automatically on save so that noone has to think about formatting ever again.

kiddi replied:
Awesome, thanks.

johannth commented on src/Components/IntroPage.jsx:
```diff
@@ -0,0 +1,27 @@
+  );
+};
+
+export default IntroPage;
```
This is definitely nice but with smaller projects I like to keep things in a single file until I've gotten the whole to make sense. Breaking it too early into separate files when the project is very small introduces mental overhead imo.

kiddi replied:
Roger.

johannth commented on src/App.css:
```diff
@@ -1,5 +1,83 @@
+
+.DonateFormPage1__wrapper {
+  display: flex;
+  width: 100vw;
```
This is neat. Didn't know this existed.

johannth committed:
Inline IntroPage + css and use box-sizing instead of -webkit/-moz-box-sizing

johannth reviewed (commented):
Nice initiative, much better mobile styling. Mildly disagree with code split at this point (because nothing works yet - we might still refactor hugely)

kiddi committed:
Merge pull request #2 from johannth/styling-feedback

PR review change to styling branch

kiddi committed:
more styling

kiddi committed:
removing unused code

kiddi commented:
- Added basic styling to donation form page 2
- Added dynamic progressbar
- Added viewport wrapper, to make it not look bad when it's in desktop view
- Moved `App ... render()` to end of file, to stop ES-Lint warning
- Added 0 tests :trollface:
